app/src/main/java/id_iot/demo/CollectionAdapter.java

```diff
@@ -0,0 +1,102 @@
+package id_iot.demo;
+
+import android.content.Context;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Filterable;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Filter;
+import android.widget.Filterable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CollectionAdapter extends BaseAdapter implements Filterable {
+    private Context mContext;
+    private ArrayList<Recipe> recipes;
+    private ArrayList<Recipe> backupRecipes;
+    ValueFilter valueFilter;
+
+    public CollectionAdapter(Context c, ArrayList<Recipe> recipes) {
```

Comment:
first variable use shortened name - just c, second is full name - recipes - this is again really inconsistent.